### PR TITLE
tropics converter update

### LINKED ISSUE #1354: TROPICS converter update for newer calibration
## Description

update the TROPICS converter for pathfinder data with new calibration

## Requirements

ioda python interface TROPICS data from NASA DISC 

## Dependencies
PR #1355 
PR  #1358

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

this updates the TROPICS converter for L1B files v3 and maintains backward support for v2

## Issue(s) addressed

Resolves #1354 

## Checklist

- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by BenjaminRuston** [comment]
@mjkagnes123 finished the update of the keys for the files version 3.04.x something like that I think,, but the converter now works for these files.  Please give it a try you should be able to switch to the branch and give it a shot I just ran it like so:

```
python3 /discover/nobackup/bruston/work/JCSDA/JEDI/ioda-bundle-intel/iodaconv/src/hdf5/tropics_2ioda.py -i /discover/nobackup/bruston/data_repos/tropics/nasa_disc/20220216/TROPICS01.BRTT.L1B.Orbit03504.V03-04.ST20220216-012443.ET20220216-025955.CT20230303-224239.nc -d 2022021600 -o ./test.nc4
```

**Comment by mjkagnes123** [comment]
I tested the updated tropics_2ioda.py in discover with TROPICS data and the code works well  for old version and new version of data.

**Review by mjkagnes123 — APPROVED** [review]
Updated code works well with latest version of TROPICS L1B data set.

**Comment by smaticka** [comment]
@BenjaminRuston I don't have access to discover. If you copy the file(s) to orion, I can test it.

File: src/hdf5/tropics_2ioda.py [hunk 1]
```diff
@@ -35,11 +35,16 @@
 obsValName = iconv.OvalName()
 
 # globals
-TROPICS01_WMO_sat_ID = 691
+TROPICS01_WMO_sat_ID = 933  # 691  did this ID get updated by WMO
```
Comment (BenjaminRuston) [inline src/hdf5/tropics_2ioda.py:38]: @BenjaminTJohnson did you see this?  the WMO ID for the TROPICS pathfinder apparently changed I got this from WMO OSCAR page when I was looking up the TROPICS-06 and -07

are these IDs in your coefficient files?

**Comment by smaticka** [comment]
When I run with the example file on Orion, I get an error that looks like possibly the `float` `int` problem. This is the error:

```
(venv) Orion-login-2: /tropics >source run_gmi_converter.sh 
Traceback (most recent call last):
  File "/work2/noaa/da/smaticka/ioda_bundle_26june_intel/ioda-bundle/iodaconv/src/hdf5/tropics_2ioda.py", line 27, in <module>
    from pyiodaconv.def_jedi_utils import ioda_int_type, ioda_float_type, epoch
ImportError: cannot import name 'ioda_int_type' from 'pyiodaconv.def_jedi_utils' (/work2/noaa/da/smaticka/ioda_bundle_26june_intel/build/lib/python3.9/pyiodaconv/def_jedi_utils.py)
```
After running the converter using this command:

`python3 /work2/noaa/da/smaticka/ioda_bundle_26june_intel/ioda-bundle/iodaconv/src/hdf5/tropics_2ioda.py -i /work2/noaa/jcsda/bruston/data_repos/tropics/nasa_disc/20220216/TROPICS01.BRTT.L1B.Orbit03504.V03-04.ST20220216-012443.ET20220216-025955.CT20230303-224239.nc -d 2022021600 -o ../ioda/tropics/test.nc4`

The following variables were `float` in the example file, should they be `int`?:

`lunar_zenith_angle`, `lunar_azimuth_angle`, `solar_zenith_angle`, `solar_azimuth_angle`, `scPosECEF`, `scQuatECEF`, `instrTemp`, `NEDT_DS`, `NEDT_ND`.

**Review by smaticka — APPROVED** [review]
Thanks @BenjaminRuston ! 

This works for me on orion. (I just need to rebuild my bundle to get the changes in the `def_jedi_utils.py` script.

**Pull request merged**